GreetingTest.java

```diff
@@ -0,0 +1,28 @@
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.After;
+import static org.junit.Assert.assertEquals;
+
+public class GreetingTest {
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    public void setUp() {
+        System.setOut(new PrintStream(outContent));
+    }
+
+    @After
+    public void tearDown() {
+        System.setOut(null);
+    }
+    
+    @Test
+    public void testV1() {
+        Greeting greeting = new Greeting();
+        greeting.sayHello();
```

Comment:
This is okay, but just FYI you can call the main method directly yourself here instead of using an instance method:
Greeting.main(null);
which assumes you are not using the argument...